### PR TITLE
Correct errors in the API spec

### LINKED ISSUE #11: Make the API specs useful for API client generation with OpenAPI Generator
As it stands right now, the Lichess API specs can not be used to generate an API with OpenAPI Generator. The API specs seem to be malformed in various ways:

- Responses seem to require a `description`
- `security: None` doesn't work correctly and can be replaced with `security: []` and `components.securitySchemes.None` removed as per [OpenAPI documentation](https://swagger.io/docs/specification/authentication/)
- A specification for the path parameter `userName` of `/games/export/{username}` is missing

Furthermore, even when fixing these issues, all response objects properties (save `BotGameState`) aren't actually described in the API, those response objects merely hold an example JSON response.

There might be more issues yet unexplored, I'm not very proficient in OpenAPI.

If this issue were fixed, it would make using the API much easier.

---

OpenAPI Generator output when trying to generate a Java API for Lichess:

      PS B:\Marvin\Desktop\openapi-generator> java -jar .\modules\openapi-generator-cli\target\openapi-generator-cli.jar generate -i https://raw.githubusercontent.com/lichess-org/api/master/doc/specs/lichess-api.yaml -g java -o lichess-java
      Exception in thread "main" org.openapitools.codegen.SpecValidationException: There were issues with the specification. The option can be disabled via validateSpec (Maven/Gradle) or --skip-validate-spec (CLI).
       | Error count: 49, Warning count: 1
      Errors:
              -attribute paths.'/api/games/user/{username}'(get).responses.200.description is missing
              -attribute paths.'/api/bot/account/upgrade'(post).responses.400.description is missing
              -attribute paths.'/api/stream/event'(get).responses.200.description is missing
              -attribute paths.'/api/challenge/{username}'(post).responses.200.description is missing
              -attribute paths.'/api/account/kid'(post).responses.200.description is missing
              -attribute paths.'/api/users/status'(get).responses.200.description is missing
              -attribute paths.'/api/account'(get).responses.200.description is missing
              -attribute paths.'/api/user/{username}/rating-history'(get).responses.200.description is missing
              -attribute components.securitySchemes.None.type.type is not of type `http|apiKey|oauth2|openIdConnect `
              -attribute paths.'/api/bot/account/upgrade'(post).responses.200.description is missing
              -attribute paths.'/api/challenge/{challengeId}/accept'(post).responses.200.description is missing
              -attribute paths.'/api/user/{username}/followers'(get).responses.200.description is missing
              -attribute paths.'/team/{teamId}/users'(get).responses.200.description is missing
              -attribute paths.'/api/tournament'(post).responses.200.description is missing
              -attribute paths.'/tv/channels'(get).responses.200.description is missing
              -attribute paths.'/api/bot/game/{gameId}/move/{move}'(post).responses.200.description is missing
              -attribute paths.'/api/bot/game/{gameId}/resign'(post).responses.400.description is missing
              -attribute paths.'/api/bot/game/{gameId}/move/{move}'(post).responses.400.description is missing
              -attribute paths.'/api/bot/game/{gameId}/resign'(post).responses.200.description is missing
              -attribute paths.'/api/tournament'(post).responses.400.description is missing
              -attribute paths.'/api/challenge/{challengeId}/decline'(post).responses.404.description is missing
              -attribute paths.'/api/account/playing'(get).responses.200.description is missing
              -attribute paths.'/games/export/{username}'. Declared path parameter username needs to be defined as a path parameter in path or operation level
              -attribute paths.'/api/user/{username}'(get).responses.200.description is missing
              -attribute paths.'/api/bot/game/stream/{gameId}'(get).responses.200.description is missing
              -attribute paths.'/api/account/email'(get).responses.200.description is missing
              -attribute paths.'/api/challenge/{challengeId}/decline'(post).responses.200.description is missing
              -attribute paths.'/streamer/live'(get).responses.200.description is missing
              -attribute paths.'/api/account/kid'(get).responses.200.description is missing
              -attribute paths.'/api/tournament/{id}/results'(get).responses.200.description is missing
              -attribute paths.'/api/bot/game/{gameId}/abort'(post).responses.400.description is missing
              -attribute paths.'/game/export/{gameId}'(get).responses.200.description is missing
              -attribute paths.'/api/stream/games-by-users'(post).responses.200.description is missing
              -attribute paths.'/api/bot/game/{gameId}/chat'(post).responses.200.description is missing
              -attribute paths.'/api/account/preferences'(get).responses.200.description is missing
              -attribute paths.'/games/export/_ids'(post).responses.200.description is missing
              -attribute paths.'/player/top/{nb}/{perfType}'(get).responses.200.description is missing
              -attribute paths.'/api/users'(post).responses.200.description is missing
              -attribute paths.'/api/bot/game/{gameId}/chat'(post).responses.400.description is missing
              -attribute paths.'/player'(get).responses.200.description is missing
              -attribute paths.'/games/export/{username}'(get).responses is missing
              -attribute paths.'/api/user/{username}/following'(get).responses.200.description is missing
              -attribute paths.'/api/challenge/{challengeId}/accept'(post).responses.404.description is missing
              -attribute paths.'/api/tournament/{id}/games'(get).responses.200.description is missing
              -attribute paths.'/api/bot/game/{gameId}/abort'(post).responses.200.description is missing
              -attribute paths.'/api/users/titled'(get).responses.200.description is missing
              -attribute paths.'/api/tournament'(get).responses.200.description is missing
              -attribute paths.'/api/challenge/{username}'(post).responses.400.description is missing
              -attribute paths.'/api/user/{username}/activity'(get).responses.200.description is missing
      Warnings:
              -attribute paths.'/api/games/user/{username}'(get).responses.200.description is missing
              -attribute paths.'/api/bot/account/upgrade'(post).responses.400.description is missing
              -attribute paths.'/api/stream/event'(get).responses.200.description is missing
              -attribute paths.'/api/challenge/{username}'(post).responses.200.description is missing
              -attribute paths.'/api/account/kid'(post).responses.200.description is missing
              -attribute paths.'/api/users/status'(get).responses.200.description is missing
              -attribute paths.'/api/account'(get).responses.200.description is missing
              -attribute paths.'/api/user/{username}/rating-history'(get).responses.200.description is missing
              -attribute components.securitySchemes.None.type.type is not of type `http|apiKey|oauth2|openIdConnect `
              -attribute paths.'/api/bot/account/upgrade'(post).responses.200.description is missing
              -attribute paths.'/api/challenge/{challengeId}/accept'(post).responses.200.description is missing
              -attribute paths.'/api/user/{username}/followers'(get).responses.200.description is missing
              -attribute paths.'/team/{teamId}/users'(get).responses.200.description is missing
              -attribute paths.'/api/tournament'(post).responses.200.description is missing
              -attribute paths.'/tv/channels'(get).responses.200.description is missing
              -attribute paths.'/api/bot/game/{gameId}/move/{move}'(post).responses.200.description is missing
              -attribute paths.'/api/bot/game/{gameId}/resign'(post).responses.400.description is missing
              -attribute paths.'/api/bot/game/{gameId}/move/{move}'(post).responses.400.description is missing
              -attribute paths.'/api/bot/game/{gameId}/resign'(post).responses.200.description is missing
              -attribute paths.'/api/tournament'(post).responses.400.description is missing
              -attribute paths.'/api/challenge/{challengeId}/decline'(post).responses.404.description is missing
              -attribute paths.'/api/account/playing'(get).responses.200.description is missing
              -attribute paths.'/games/export/{username}'. Declared path parameter username needs to be defined as a path parameter in path or operation level
              -attribute paths.'/api/user/{username}'(get).responses.200.description is missing
              -attribute paths.'/api/bot/game/stream/{gameId}'(get).responses.200.description is missing
              -attribute paths.'/api/account/email'(get).responses.200.description is missing
              -attribute paths.'/api/challenge/{challengeId}/decline'(post).responses.200.description is missing
              -attribute paths.'/streamer/live'(get).responses.200.description is missing
              -attribute paths.'/api/account/kid'(get).responses.200.description is missing
              -attribute paths.'/api/tournament/{id}/results'(get).responses.200.description is missing
              -attribute paths.'/api/bot/game/{gameId}/abort'(post).responses.400.description is missing
              -attribute paths.'/game/export/{gameId}'(get).responses.200.description is missing
              -attribute paths.'/api/stream/games-by-users'(post).responses.200.description is missing
              -attribute paths.'/api/bot/game/{gameId}/chat'(post).responses.200.description is missing
              -attribute paths.'/api/account/preferences'(get).responses.200.description is missing
              -attribute paths.'/games/export/_ids'(post).responses.200.description is missing
              -attribute paths.'/player/top/{nb}/{perfType}'(get).responses.200.description is missing
              -attribute paths.'/api/users'(post).responses.200.description is missing
              -attribute paths.'/api/bot/game/{gameId}/chat'(post).responses.400.description is missing
              -attribute paths.'/player'(get).responses.200.description is missing
              -attribute paths.'/games/export/{username}'(get).responses is missing
              -attribute paths.'/api/user/{username}/following'(get).responses.200.description is missing
              -attribute paths.'/api/challenge/{challengeId}/accept'(post).responses.404.description is missing
              -attribute paths.'/api/tournament/{id}/games'(get).responses.200.description is missing
              -attribute paths.'/api/bot/game/{gameId}/abort'(post).responses.200.description is missing
              -attribute paths.'/api/users/titled'(get).responses.200.description is missing
              -attribute paths.'/api/tournament'(get).responses.200.description is missing
              -attribute paths.'/api/challenge/{username}'(post).responses.400.description is missing
              -attribute paths.'/api/user/{username}/activity'(get).responses.200.description is missing
  
              at org.openapitools.codegen.config.CodegenConfigurator.toClientOptInput(CodegenConfigurator.java:566)
              at org.openapitools.codegen.cmd.Generate.run(Generate.java:353)
              at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:61)

### PR DESCRIPTION
**Pull request by Pamplemousse**
Fixes #11!

`openapi-generator-cli validate -i doc/specs/lichess-api.yaml` gives the remaining spec violations.

Signed-off-by: Pamplemousse <xav.maso@gmail.com>

### CODE REVIEW

File: doc/specs/lichess-api.yaml [hunk 1]
```diff
@@ -174,6 +174,7 @@ paths:
           example: aliquantus,chess-network,lovlas
       responses:
         200:
+          description: The list of user and their respective statuses.
```
Comment (trollepierre) [inline doc/specs/lichess-api.yaml:177]:
```suggestion
          description: The list of users and their respective statuses.
```

**Comment by Pamplemousse** [comment]
This is a best effort: I never worked with OpenAPI before, and it's my first contribution to Lichess...
I tried to fix the errors in a clear and idiomatic way, while doing my best to keep the original intent of the spec.
My contribution is split into several commits to separate issues that were "trivial" from the ones that might raise discussions / rejection.

cc @ornicar @niklasf (and @markv for the original issue)

**Comment by sjamesr** [comment]
Thank you so much, this is great! If I'm not mistaken, the `fen` parameter in `tablebaseStandard` is missing a type around line 5010. It should read:

```yaml
  /standard:
    get:
      operationId: tablebaseStandard
      summary: Tablebase lookup
      description: |
        **Endpoint: https://tablebase.lichess.ovh**

        Runs https://github.com/niklasf/lila-tablebase with `--sloppy-real-wdl`.

        Example: `curl http://tablebase.lichess.ovh/standard?fen=4k3/6KP/8/8/8/8/7p/8_w_-_-_0_1`
      tags:
        - Tablebase
      security: []
      parameters:
        - in: query
          name: fen
          description: FEN of the position. Underscores allowed. The halfmove clock is taken into account for WDL values.
          schema:          # <-----
            type: string   # <----- these two lines are missing
      responses:
        200:
          description: The tablebase information for the position in standard chess.
          headers:
            Access-Control-Allow-Origin:
              schema:
                type: string
                default: "'*'"
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/TablebaseJson'
```

**Comment by Pamplemousse** [comment]
@sjamesr Hey! How do you know this type is missing?

I don't see any more errors when running `openapi-generator-cli validate -i doc/specs/lichess-api.yaml`...
And it's not amongst the warnings either.

**Comment by sjamesr** [comment]
Without it, using the default Java generator, the `fen` parameter would not be emitted correctly. I'll try reproduce the problem for you.

**Comment by Pamplemousse** [comment]
Interesting...
I would say that there might be a bug in either `openapi-generator-cli validate` or "the default Java generator" (are you referring to this: https://github.com/OpenAPITools/openapi-generator ?).

**Comment by ornicar** [comment]
Nice. I really wish we had a way to get rid of the duplication, such as

```
          headers:
            Access-Control-Allow-Origin:
              schema:
                type: string
                default: "'*'"
```

There is so much duplicated stuff in there, and I think it's the fault of the format.

Maybe we should be generating it with an actual programming language instead?

**Comment by Pamplemousse** [comment]
@ornicar 
Yeah, I looked into that, the OpenAPI format does not allow such modularity: see OAI/OpenAPI-Specification#690 .

Even if it had a default value, or was declared in a unique place, each response would need to declare a reference to this unique location... The duplication would just be "shifted".

**Pull request merged**